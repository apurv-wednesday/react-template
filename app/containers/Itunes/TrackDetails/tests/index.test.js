import React from 'react';
import { renderProvider } from '@utils/testUtils';
import { TrackDetails } from '../index';

describe('<TrackDetails /> container tests', () => {
  it('should render and match the snapshot', () => {
    const mockProps = {
      dispatchRequestTrackDetails: jest.fn(),
      trackDetails: {
        trackName: 'Test Track',
        artistName: 'Test Artist',
        previewUrl: 'https://example.com/preview',
        artworkUrl100: 'https://example.com/artwork',
        collectionName: 'Test Collection'
      }
    };
    const { baseElement } = renderProvider(<TrackDetails {...mockProps} />);
    expect(baseElement).toMatchSnapshot();
  });
  it('should render null when trackDetails is null', () => {
    const mockProps = {
      dispatchRequestTrackDetails: jest.fn(),
      trackDetails: null
    };
    const { baseElement } = renderProvider(<TrackDetails {...mockProps} />);
    expect(baseElement).toMatchSnapshot();
  });
});
