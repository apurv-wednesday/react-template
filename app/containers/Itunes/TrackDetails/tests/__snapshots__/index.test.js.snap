// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackDetails /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root e1a3jrlg0 css-15kwy1v-MuiPaper-root-MuiCard-root"
      data-testid="track-component"
    >
      <div
        class="MuiBox-root css-j7qwjs"
      >
        <div
          class="MuiCardContent-root css-1qaznj3-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            Test Track
          </div>
          <div
            class="MuiTypography-root MuiTypography-subtitle1 css-ecpxie-MuiTypography-root"
          >
            Test Artist
          </div>
        </div>
        <div
          class="MuiBox-root css-1ndgitk"
        >
          <button
            aria-label="play/pause"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="play-pause-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PlayArrowIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 5v14l11-7z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-114os8-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Show Details
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <img
        alt="Test Collection"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1rorjgf-MuiCardMedia-root"
        src="https://example.com/artwork"
      />
      <audio
        data-testid="trackAudio"
        src="https://example.com/preview"
      />
    </div>
  </div>
</body>
`;

exports[`<TrackDetails /> container tests should render null when trackDetails is null 1`] = `
<body>
  <div />
</body>
`;
